src/Microsoft.Framework.Runtime/DependencyManagement/NuGetDependencyResolver.cs

```diff
@@ -106,6 +106,19 @@ public LibraryDescription GetDescription(LibraryRange libraryRange, FrameworkNam
                         resolved = false;
                         dependencies = Enumerable.Empty<LibraryDependency>();
                     }
+
+                    // If a NuGet dependency is supposed to provide assemblies but there is no assembly compatible with
+                    // current target framework, we should mark this dependency as unresolved
+                    if (targetLibrary != null)
+                    {
+                        var containsAssembly = package.LockFileLibrary.Files
+                            .Any(x => x.StartsWith($"ref{Path.DirectorySeparatorChar}") ||
```

Comment:
It is in lock file on disk. We convert `/` to `Path.Separator` when we read a lock file into memory:
https://github.com/aspnet/dnx/blob/dev/src/Microsoft.Framework.PackageManager/Restore/LockFileFormat.cs#L367

I agree that we need really good tests. But the code here is correct.